src/resources/resources.service.ts

```diff
@@ -58,4 +58,25 @@ export class ResourcesService {
 
     return { message: "Resource deleted successfully" };
   }
+
+  async updateResource(
+    resourceId: number,
+    updateResourceDto: UpdateResourceDto,
+    file?: Express.Multer.File,
+  ) {
+    const resource = await this.resourcesRepository.findOneOrFail(resourceId);
+    if (file) {
+      const oldFileKey = resource.fileUrl.split("project-dev-bucket/")[1];
+      await this.fileUploadsService.deleteFromS3(oldFileKey);
```

Comment:
Probably best to have checks here for error cases

Reply:
Already addressed the issue on the previous PR. Used try catch block on the fileUpload.service layer